### PR TITLE
Feature/sg 1039 add factory doc

### PR DESCRIPTION
**Pull request by Louis-Dupont**
Goal: Clarify what are factories and how to use them

### CODE REVIEW

File: documentation/source/factories.md [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+        prob: 0.7 
+```
+
+Final Step: Ensure that you import the module containing `MyTransformName` into your script. 
+Doing so will trigger the registration function, allowing SuperGradients to recognize it.
+
+Here is an example (adapted from the [train_from_recipe script](https://github.com/Deci-AI/super-gradients/blob/master/src/super_gradients/train_from_recipe.py)).
+
+```python
+from .my_module import MyTransform # Importing the module is enough as it will trigger the register_model function
```
Comment (dafna-r) [inline documentation/source/factories.md:107]: should be "register_transform" instead of "register_model"

File: documentation/source/factories.md [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+    register_unet_backbone_stage
+    register_unet_up_block
+    register_target_generator
+    register_lr_scheduler
+    register_lr_warmup
+    register_sg_logger
+    register_collate_function
+    register_sampler
+    register_optimizer
+    register_processing
```
Comment (dafna-r) [inline documentation/source/factories.md:205]: add comma ","

**Review by dafna-r — COMMENTED** [review]
small typo corrections

File: documentation/source/factories.md (hunk 2 above)
Comment (Louis-Dupont) [reply documentation/source/factories.md]: Fixed in [6b8e8ee](https://github.com/Deci-AI/super-gradients/pull/1395/commits/6b8e8ee8aee63068e1aee19cef7b314c06a44373)

File: documentation/source/factories.md (hunk 1 above)
Comment (Louis-Dupont) [reply documentation/source/factories.md]: Fixed in [6b8e8ee](https://github.com/Deci-AI/super-gradients/pull/1395/commits/6b8e8ee8aee63068e1aee19cef7b314c06a44373)

**Review by BloodAxe — APPROVED** [review]
LGTM
